### PR TITLE
tikv: add new config for auto adjust read pool thread count

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by glorv** [comment]
depending on https://github.com/tikv/tikv/pull/13314 to be merged first.

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -247,6 +247,11 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 + 默认值：2000
 + 最小值：2
 
+### `auto-adjust-pool-size` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
++ 是否开启自动基于当前的 CPU 使用情况调整线程池的线程数量。开启此配置可以通过自动调整统一处理读请求的线程池的线程数量优化 TiKV 的读性能。目前线程池自动调整的范围为：`[max-thread-count, CPU]`。
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:252]:
```suggestion
+ 是否开启自动调整线程池的线程数量。开启此配置可以基于当前的 CPU 使用情况，自动调整统一处理读请求的线程池的线程数量，优化 TiKV 的读性能。目前线程池自动调整的范围为：`[max-thread-count, CPU 线程数]`。
```

Comment (Oreoxmt) [reply tikv-configuration-file.md:252]: 1. auto-adjust-pool-size 支持在线修改吗，如果支持，需要在[在线修改 TiKV 配置](https://docs.pingcap.com/zh/tidb/stable/dynamic-config#%E5%9C%A8%E7%BA%BF%E4%BF%AE%E6%94%B9-tikv-%E9%85%8D%E7%BD%AE)中添加这个配置项
2. 是否需要修改其它文档，例如 [TiKV 线程池大小配置](https://docs.pingcap.com/zh/tidb/stable/three-nodes-hybrid-deployment#tikv-%E7%BA%BF%E7%A8%8B%E6%B1%A0%E5%A4%A7%E5%B0%8F%E9%85%8D%E7%BD%AE)、[TiKV 线程池性能调优](https://docs.pingcap.com/zh/tidb/stable/tune-tikv-thread-performance#tikv-%E7%BA%BF%E7%A8%8B%E6%B1%A0%E6%80%A7%E8%83%BD%E8%B0%83%E4%BC%98)

Comment (glorv) [reply tikv-configuration-file.md:252]: 前面其他的地方都使用 `CPU` 表示 cpu quota， 这里单独用 `CPU 线程数`有点奇怪。

Comment (glorv) [reply tikv-configuration-file.md]: fixed. @Oreoxmt PTAL again

File: tune-tikv-thread-performance.md [hunk 2]
```diff
@@ -63,6 +63,8 @@ TiKV 的读取请求分为两类：
 
     通常建议根据业务负载特性调整其 CPU 使用率在线程池大小的 60%～90% 之间（如果用户 Grafana 上 TiKV-Details.Thread CPU.Unified read pool CPU 的峰值不超过 800%，那么建议用户将 `readpool.unified.max-thread-count` 设置为 10，过多的线程数会造成更频繁的线程切换，并且抢占其他线程池的资源）。
 
+    TiKV 从 v6.3.0 开始支持根据统一读线程池（UnifyReadPool）的 CPU 利用率自动动态调整线程池的线程数量，可以通过设置 `readpool.unified.auto-adjust-pool-size = true` 开启此功能。对于重读并且峰值 CUP 利用率超过 80% 的集群，建议开启此配置。
```
Comment (Oreoxmt) [inline tune-tikv-thread-performance.md:66]:
```suggestion
    TiKV 从 v6.3.0 开始支持根据统一读线程池（UnifyReadPool）的 CPU 利用率自动动态调整线程池的线程数量，可以通过设置 [`readpool.unified.auto-adjust-pool-size = true`](/tikv-configuration-file.md#auto-adjust-pool-size-从-v630-版本开始引入) 开启此功能。对于重读并且峰值 CPU 利用率超过 80% 的集群，建议开启此配置。
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -247,6 +247,11 @@ TiKV 配置文件比命令行参数支持更多的选项。你可以在 [etc/con
 + 默认值：2000
 + 最小值：2
 
+### `auto-adjust-pool-size` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
++ 是否开启自动调整线程池的大小。开启此配置可以基于当前的 CPU 使用情况，自动调整统一处理读请求的线程池 (UnifyReadPool) 的大小，优化 TiKV 的读性能。目前线程池自动调整的范围为：`[max-thread-count, MAX(4, CPU)]`(上限与 `max-thread-count` 可设置的最大值相同)。
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:252]:
```suggestion
+ 是否开启自动调整线程池的大小。开启此配置可以基于当前的 CPU 使用情况，自动调整统一处理读请求的线程池 (UnifyReadPool) 的大小，优化 TiKV 的读性能。目前线程池自动调整的范围为：`[max-thread-count, MAX(4, CPU)]`(上限与 [`max-thread-count`](#max-thread-count) 可设置的最大值相同)。
```

File: dynamic-config.md [hunk 4]
```diff
@@ -156,6 +156,7 @@ show warnings;
 | raftstore.apply-max-batch-size | Raft 状态机由 BatchSystem 批量执行数据写入请求，该配置项指定每批可执行请求的最多 Raft 状态机个数。 |
 | raftstore.store-max-batch-size |  Raft 状态机由 BatchSystem 批量执行把日志落盘至磁盘的请求，该配置项指定每批可执行请求的最多 Raft 状态机个数。 |
 | readpool.unified.max-thread-count | 统一处理读请求的线程池最多的线程数量，即 UnifyReadPool 线程池大小 |
+| readpool.unified.auto-adjust-pool-size | 是否开启自适应调整统一处理读请求的线程池的线程数量 |
```
Comment (Oreoxmt) [inline dynamic-config.md:159]:
```suggestion
| readpool.unified.auto-adjust-pool-size | 是否开启自适应调整 UnifyReadPool 的大小 |
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: tune-tikv-thread-performance.md [hunk 5]
```diff
@@ -63,6 +63,8 @@ TiKV 的读取请求分为两类：
 
     通常建议根据业务负载特性调整其 CPU 使用率在线程池大小的 60%～90% 之间（如果用户 Grafana 上 TiKV-Details.Thread CPU.Unified read pool CPU 的峰值不超过 800%，那么建议用户将 `readpool.unified.max-thread-count` 设置为 10，过多的线程数会造成更频繁的线程切换，并且抢占其他线程池的资源）。
 
+    TiKV 从 v6.3.0 开始支持根据统一读线程池（UnifyReadPool）的 CPU 利用率自动动态调整线程池的线程数量，可以通过配置 [`readpool.unified.auto-adjust-pool-size`](/tikv-configuration-file.md#auto-adjust-pool-size-从-v630-版本开始引入) 开启此功能。对于重读并且峰值 CPU 利用率超过 80% 的集群，建议开启此配置。
```
Comment (ran-huang) [inline tune-tikv-thread-performance.md:66]:
```suggestion
    TiKV 从 v6.3.0 开始支持根据统一读线程池 (UnifyReadPool) 的 CPU 利用率自动动态调整线程池的线程数量，可以通过配置 [`readpool.unified.auto-adjust-pool-size`](/tikv-configuration-file.md#auto-adjust-pool-size-从-v630-版本开始引入) 开启此功能。对于重读并且峰值 CPU 利用率超过 80% 的集群，建议开启此配置。
```

**Review by ran-huang — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Pull request merged**